packages/dds/tree/docs/wip/Modules.md

```diff
@@ -0,0 +1,196 @@
+# Modules
+
+There are a wide variety of use-cases for a Fluid tree, and there are several trade-offs in the design space that should be made differently for the different use-cases.
+To enable flexibility both for how our users configure the system, as well as when we modify the implementation,
+the implementation consist of collection of modules which can be composed in a few different ways to address a wide variety of use cases.
+
+This breakdown is intended as an architectural approach to ensure we can deliver functionality in an incremental way (minimizing the work needed for the early milestones),
+as well as quickly respond to specific feature requests, and perform targeted unit testing.
+
+These module boundaries could additionally be exposes as public packages,
+allowing our consumers to also extend and configure the functionality in these ways (increasing our internal API stability obligations),
```

Comment:
The primary motivations for package boundaries are discoverability, versioning, and breaking cyclic dependencies.

Sometimes bundle size is a consideration, although tree shaking should be the first line of defense.

Coarse layering is a consideration too, but usually because of one of the three primary motivations.

Reply:
I have rephrased this to be less specific.